cluster/src/dunit/scala/io/snappydata/cluster/StringAsVarcharDUnitTest.scala

```diff
@@ -0,0 +1,183 @@
+    executeQuery("hint", true)
+  }
+
+  def testSelectJoinQueryWithSQLConf(): Unit = {
+    executeQuery("sqlConf", true)
+  }
+
+  def executeQuery(conf: String, join: Boolean = false): Unit = {
+    var stringType = "VARCHAR"
+    val netPort1 = AvailablePortHelper.getRandomAvailableTCPPort
+    vm2.invoke(classOf[ClusterManagerTestBase], "startNetServer", netPort1)
+    val conn = getANetConnection(netPort1)
+
+    createTableAndInsertData(conn)
+
+    val s = conn.createStatement()
+    var (query, count, cols, tName) = join match {
+      case true => (s"select t1.col_int, t1.col_string, t2.col_varchar, t2.col_clob, t2.col_char " +
+          s"from $tableName1 t1, $tableName2 t2 where t1.col_int = t2.col_int", 2, 5, "T1")
+      case false => (s"select * from $tableName2", 2, 5, tableName2)
+    }
+    conf match {
+      case "none" =>
+      case "hint" => query += " --+ stringAsClob(1)"
+        stringType = "CLOB"
+      case "sqlConf" => s.execute("set spark.sql.stringAsClob=true")
+        stringType = "CLOB"
+      case _ => throw new IllegalArgumentException(s"Invalid argument $conf")
+    }
+
+    s.executeQuery(query)
+    val rs = s.getResultSet
+    var rows = 0
+    while (rs.next()) {
+      rows += 1
+      if (join) {
+        if (rs.getInt(1) == 1) {
+          assert(rs.getString(2).equals("t1.1.string"), s"actual string ${rs.getString(2)}")
+          assert(rs.getString(4).equals("t2.1.clob"), s"actual clob ${rs.getString(4)}")
+        } else {
+          assert(rs.getInt(1) == 4, s"actual int ${rs.getInt(1)}")
+          assert(rs.getString(2).equals("t1.4.string"), s"actual string ${rs.getString(2)}")
+          assert(rs.getString(3).equals("t2.4.varchar"), s"actual varchar ${rs.getString(3)}")
+        }
+      } else {
+        if (rs.getInt(1) == 1) {
+          assert(rs.getString(2).equals("t2.1.string"), s"actual string ${rs.getString(2)}")
+          assert(rs.getString(3).equals("t2.1.varchar"), s"actual varchar ${rs.getString(3)}")
+        } else {
+          assert(rs.getInt(1) == 4, s"actual int ${rs.getInt(1)}")
+          assert(rs.getString(2).equals("t2.4.string"), s"actual string ${rs.getString(2)}")
+          assert(rs.getString(4).equals("t2.4.clob"), s"actual clob ${rs.getString(4)}")
+        }
+      }
+    }
+    assert(rows == count)
+
+    val md = rs.getMetaData
+    assert(md.getColumnCount == cols)
+    logInfo("metadata col cnt = " + md.getColumnCount)
+    var i = 0
+    while (i < cols) {
+      i += 1
+      logInfo("col name = " + md.getColumnName(i) +
+          ", col type " + md.getColumnTypeName(i) + ", col table name = " + md.getTableName(i))
+    }
+    assert(md.getColumnName(1).equals("COL_INT"))
+    assert(md.getColumnTypeName(1).equals("INTEGER"))
+
+    assert(md.getColumnName(2).equals("COL_STRING"))
+    assert(md.getColumnTypeName(2).equals(stringType))
+
+    assert(md.getColumnName(3).equals("COL_VARCHAR"))
+    assert(md.getColumnTypeName(3).equals("VARCHAR"))
```

Comment:
Test the size also from the metadata. You can use:

ResultSetMetaData.getPrecision ( jdbc api )